packages/apple-targets/src/withWidget.ts

```diff
@@ -318,6 +336,11 @@ const withWidget: ConfigPlugin<Props> = (config, props) => {
     },
   ]);
 
+  if (props.infoPlist) {
```

Comment:
I think this feature should be more clever. If the `infoPlist` is defined then it should generate a new infoPlist each time and put it in a generated folder. We don't have a pattern for this yet, e.g. entitlements should do the same.